### DISCUSSION TITLE
Resource container debugging: Is there a way to attach to resource type container?

### DISCUSSION THREAD

**Discussion by maestrow** [concourse/concourse#7486] [Help & Support] [answered]
Just now I faced in [weird problem](https://github.com/concourse/git-resource/issues/369) with git-resource: `git_config` option doesn't set global git parameters. I run tests from [git-resource](https://github.com/concourse/git-resource) repo myself and check one specific test `get.sh/it_can_get_and_set_git_config`. And it works well. So I'm wondered what is the reason not to create global git configuration in my case? 

I'm looking for some debugging techniques. 

- Is there a way not to remove resource type container after it finishes working?
- Is there a way to attach to that container?

**Comment by taylorsilva** [comment, marked as answer]
You can `fly intercept` into check containers with the `--check=PIPELINE/CHECK` flag. `fly intercept` is like doing `docker exec -it <container> -- bash` (docker is not involved with this at all though).
